### PR TITLE
Replace web3modal and redux with wagmi

### LINKED ISSUE #55: Replace web3modal and redux with wagmi
https://wagmi.sh/

This library looks infinitely more modern and easy to use

### PR DESCRIPTION
**Pull request by thal0x**
It's very possible I introduced some new bugs here but I'm hoping they'll get taken care of in the next PR for properly setting up the graphql client.

When you first go to `/vault/options` there's an error that you can just close and ignore. I can't track it down, but think the next steps will take care of it.

There's also a good deal of unused code left behind, I'm going to get rid of it in a future PR.

Closes #55

### CODE REVIEW

**Comment by 0xAlcibiades** [comment]
```npm run build

> nextjs@1.0.0 build
> next build

Failed to compile.

./components/optionModal/index.tsx:4:29
Type error: Cannot find module 'react-redux' or its corresponding type declarations.

  2 | import moment from "moment";
  3 | import React, { useCallback, useEffect, useMemo } from "react";
> 4 | import { useSelector } from "react-redux";
    |                             ^
  5 | import { useAccount, useContract, useSigner } from "wagmi";
  6 |
  7 | import { smartFormatCurrency } from "../../lib/currencyFormat";
info  - Checking validity of types .%``` - the build is failing.

**Comment by thal0x** [comment]
> ```
> 
> > nextjs@1.0.0 build
> > next build
> 
> Failed to compile.
> 
> ./components/optionModal/index.tsx:4:29
> Type error: Cannot find module 'react-redux' or its corresponding type declarations.
> 
>   2 | import moment from "moment";
>   3 | import React, { useCallback, useEffect, useMemo } from "react";
> > 4 | import { useSelector } from "react-redux";
>     |                             ^
>   5 | import { useAccount, useContract, useSigner } from "wagmi";
>   6 |
>   7 | import { smartFormatCurrency } from "../../lib/currencyFormat";
> info  - Checking validity of types .%``` - the build is failing.
> ```

Should be good now

File: .github/workflows/general.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: "Valorem Frontend CI/CD"
```
Comment (0xAlcibiades) [inline .github/workflows/general.yml:1]: This resolves #50

File: components/claimModal/index.tsx [hunk 2]
```diff
@@ -19,32 +20,39 @@ function ClaimModal(props: ClaimProps): JSX.Element {
   const [hasClaim, setHasClaim] = useState<any>(null);
 
   const { claim, onClose, open } = props;
-  const wallet: Wallet = useSelector((state: SiteStore) => state.wallet);
 
-  const fetchClaimBalance = useCallback(async () => {
```
Comment (0xAlcibiades) [inline components/claimModal/index.tsx:24]: Any reason to get rid of the loading state here?

File: components/connectWalletModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                  Scan with WalletConnect to connect
+                </p>
+              </div>
+              <div className="option">
+                <img
+                  className="option__logo"
+                  src="/frame.png"
+                  alt="Frame logo"
+                />
+                <p className="option__name">Frame</p>
```
Comment (0xAlcibiades) [inline components/connectWalletModal/index.tsx:65]: Frame isn't working for now, we opened #59 to track.

File: components/claimModal/index.tsx (hunk 2 above)
Comment (thal0x) [reply components/claimModal/index.tsx:24]: It's getting loaded a different way on line 35. Unless I'm misunderstanding your question

File: components/connectWalletModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                <p className="option__text">Connect with your Frame account</p>
+              </div>
+              <div className="option" onClick={() => connect(coinbase)}>
+                <img
+                  className="option__logo"
+                  src="/coinbase-wallet.svg"
+                  alt="Coinbase Wallet logo"
+                />
+                <p className="option__name">Coinbase Wallet</p>
```
Comment (0xAlcibiades) [inline components/connectWalletModal/index.tsx:74]: This isn't working, I've opened #60 to track.

**Review by 0xAlcibiades — APPROVED** [review]
Approving with the comment that a lot of

**Pull request merged**